relay-common/src/metrics.rs

```diff
@@ -140,15 +178,51 @@ pub fn configure_statsd<A: ToSocketAddrs>(
     prefix: &str,
     host: A,
     default_tags: BTreeMap<String, String>,
+    buffering: bool,
+    sample_rate: f32,
 ) {
     let addrs: Vec<_> = host.to_socket_addrs().unwrap().collect();
     if !addrs.is_empty() {
         log::info!("reporting metrics to statsd at {}", addrs[0]);
     }
-    let statsd_client = StatsdClient::from_udp_host(prefix, &addrs[..]).unwrap();
+
+    let normalized_sample_rate = if sample_rate >= 1.0 {
+        1.0
+    } else if sample_rate <= 0.0 {
+        0.0
+    } else {
+        sample_rate
+    };
+    log::debug!(
+        "metrics sample rate is set to {}{}",
+        normalized_sample_rate,
+        if normalized_sample_rate == 0.0 {
+            ", no metrics will be reported"
+        } else {
+            ""
+        }
+    );
+
+    let socket = UdpSocket::bind("0.0.0.0:0").unwrap();
+    socket.set_nonblocking(true).unwrap();
```

Comment:
why is this needed?  This is a behaviour change from the current code which can't be undone with simply configuration.

Reply:
It's not the behavior change: https://docs.rs/cadence/0.22.0/src/cadence/client.rs.html#645
We're doing the same, just not inside the `from_udp_host` function